src/main/kotlin/org/rust/ide/hints/codeVision/RsReferenceCodeVisionProvider.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints.codeVision
+
+import com.intellij.codeInsight.codeVision.CodeVisionRelativeOrdering
+import com.intellij.codeInsight.hints.codeVision.ReferencesCodeVisionProvider
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.RsBundle
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsQualifiedNamedElement
+import org.rust.lang.core.psi.ext.searchReferences
+
+class RsReferenceCodeVisionProvider : ReferencesCodeVisionProvider() {
+    override fun acceptsFile(file: PsiFile): Boolean = file is RsFile
+
+    override fun acceptsElement(element: PsiElement): Boolean = element is RsQualifiedNamedElement
+
+    override fun getHint(element: PsiElement, file: PsiFile): String? {
+        val rsElement = element as? RsElement ?: return null
+        val usageCount = rsElement.searchReferences().count()
```

Comment:
Since the reference search can be slow, let's check that there are not so many occurrences in different files. It can be done the following way: 
```Kotlin
val searchHelper = PsiSearchHelper.getInstance(project)
val useScope = searchHelper.getUseScope(namedElement)
if (useScope is GlobalSearchScope) {
    val searchCost = searchHelper.isCheapEnoughToSearch(name, useScope, file, null)
    if (searchCost == PsiSearchHelper.SearchCostResult.TOO_MANY_OCCURRENCES) {
        return null // don't show the usages count in case of too many occurrences
    }
}
```